src/lib/annotations/AnnotationThread.js

```diff
@@ -151,20 +151,23 @@ class AnnotationThread extends EventEmitter {
         this.annotationService
             .create(annotationData)
             .then((savedAnnotation) => {
-                // If no temporary annotation is found, save to thread normally
                 const tempIdx = this.annotations.indexOf(tempAnnotation);
                 if (tempIdx === -1) {
+                    // If no temporary annotation is found, save to thread normally
                     this.saveAnnotationToThread(savedAnnotation);
+                } else {
+                    // Otherwise, replace temporary annotation with annotation saved to server
+                    this.annotations[tempIdx] = savedAnnotation;
```

Comment:
Is it possible that `this.annotations[-1]` would have `tempAnnotation` in it? Would we need to replace it with the correct one no matter what?